### DISCUSSION TITLE
How to use different MainPages/Entrypoints for different platforms?

### DISCUSSION THREAD

**Discussion by NoizeDaemon** [dotnet/maui#4194] [Q&A] [answered]
I know that I can have one XAML file handling the UI for multiple platforms, however, I've been unable to create a separate file as MainPage/entrypoint.

Let's say I want separate XAML mainpages for phones, tablets and desktops or just a windows specific XAML in the Windows-Platform folder that the app starts with when run on windows. How do I set it up? What namespaces do I use?

I've looked all over the documentation, and even though there is a whole subsection on app startup, I haven't found anything regarding that. The demo projects I've found so far all use one common starting page and only use the project specific folders for mostly non-UI-related functionality.

(I know that the main point of MAUI is to unify everything, to code less, but I think it's simpler to have different UI for desktop and phones than to cram everything into one XAML file - but I'm also just starting to learn programming, so this might be my inexperience speaking. I still want to make use of the one project approach though which is why I am trying to make MAUI work for instead of going to a more mature framework like UWP or WPF for now).

**Comment by nextfool** [comment, marked as answer]
I don't quite understand your needs here,  but seems what you need is "[conditional compilation"](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives)

Basically something like:

```
#if WINDOWS
            MainPage = new WindowsEntryPage();
#elif ANDROID
            MainPage = new AndroidEntryPage();
#elif IOS
            MainPage = new IOSEntryPage();
#endif
 
```

**Reply by NoizeDaemon** [reply]
Ah, I thought `MainPage` was connected to the `MainPage.xaml/.cs`, so I wrote sth like `MainPageWindows = new MainPageWindows()` which obviously didn't make sense to the compiler.

Now I tried again using the correct way, but ended up with a bunch of errors and missing namespaces again - because I placed the MAUI ContentPage into the Windows folder - **_placing it in the main directory_** instead works with no issues.

That raises a follow up questions:

Isn't that `MainPageWindows` then also included in all the other builds that will never use it? How can I have these ContentPages in their separate platform folders/exclude them from the other builds?

[Ideally I'd like to use the (experimental) WinUI 3 pages to design the windows specific UI, but I guess that's not the intended use of MAUI? Gotta do some more research into that, it's only loosely related to my original question which related to MAUI ContentPages, so it's probably better to open a seperate discussion if I don't find the info I need, but I'd ofc appreciate any advise in this regard as well]
